ci: Added doc linting rules to ensure google style docstrings.

KennethEnevoldsen opened the pull request:
We have specified google style docstring in the pyproject.toml, but the rule is not enforced. This PR enforces that.

Many of the changes are just autofixes by ruff.

isaac-chung reviewed (approved):
Gotta love ruff 🐶 
We saw the `tmp/results` folder before in https://github.com/embeddings-benchmark/mteb/pull/442. Is this intended?

KennethEnevoldsen commented:
> We saw the tmp/results folder before in https://github.com/embeddings-benchmark/mteb/pull/442. Is this intended?

No I have removed it. I have also removed it locally so it does not happen again
